pkg/stub/handler.go

```diff
@@ -327,6 +337,75 @@ func (h *Handler) updateCfgArch(cfg *v1.Config) *v1.Config {
 	return cfg
 }
 
+func (h *Handler) imageConfigBlocksImageStreamCreation(name string) bool {
+	//TODO openshift/client-go/config/clientset/versioned/fake and ConfigV1Interface has compile issues with
+	// respect to ConfigV1Client (missing method implementations), so for now we cannot use it in our
+	// unit tests.  In the actual runtime if we cannot create the client the operator errors out on startup.
+	if h.configclient == nil {
+		return false
+	}
+	err := wait.PollImmediate(5*time.Second, 20*time.Second, func() (done bool, err error) {
+		// if the image config allowed registry or blocked registry list will prevent the creation of imagestreams,
+		// we consider this inaccessible
+		imgCfg, err := h.configclient.Images().Get(context.TODO(), "cluster", metav1.GetOptions{})
+		if err != nil {
+			logrus.Printf("unable to retrieve image configuration as part of testing registry.redhat.io connectivity: %s", err.Error())
+			return false, nil
+		} else {
+			// check allowed domain list
+			if len(imgCfg.Spec.AllowedRegistriesForImport) > 0 {
+				for _, rl := range imgCfg.Spec.AllowedRegistriesForImport {
+					logrus.Printf("considering allowed registries domain %s for %s", rl.DomainName, name)
+					if strings.HasSuffix(name, rl.DomainName) {
+						msg := fmt.Sprintf("the allowed registries domain %s allows imagestream creation access for search param %s", rl.DomainName, name)
+						logrus.Printf(msg)
+						return true, nil
+					}
+				}
+				msg := fmt.Sprintf("no allowed registries items will permit the use of %s", name)
+				return false, fmt.Errorf(msg)
+			}
+
+			// check allowed specific registry list
+			if len(imgCfg.Spec.RegistrySources.AllowedRegistries) > 0 {
+				for _, r := range imgCfg.Spec.RegistrySources.AllowedRegistries {
+					logrus.Printf("considering allowed registry %s for %s", r, name)
+					if strings.TrimSpace(r) == strings.TrimSpace(name) {
+						msg := fmt.Sprintf("the allowed registry %s allows imagestream creation for search param %s", r, name)
+						logrus.Printf(msg)
+						return true, nil
+					}
+				}
+				msg := fmt.Sprintf("no allowed registries items will permit the use of %s", name)
+				return false, fmt.Errorf(msg)
+
+			}
+
```

Comment:
Do we need a section to check for a partial match of domain name for blocked registries also, like we do above for allowed registries?

Reply:
nope the domain thing only exists for allowed

then it is explicit registry listing for the stuff under RegistrySource